src/coreclr/vm/excep.cpp

```diff
@@ -3567,8 +3567,9 @@ LONG WatsonLastChance(                  // EXCEPTION_CONTINUE_SEARCH, _CONTINUE_
 #ifdef HOST_WINDOWS
                 CreateCrashDumpIfEnabled(fSOException);
 #endif
-                RaiseFailFastException(pExceptionInfo == NULL ? NULL : pExceptionInfo->ExceptionRecord,
-                                       pExceptionInfo == NULL ? NULL : pExceptionInfo->ContextRecord,
+                bool fAvoidReportContextToRaiseFailFast = tore.IsFatalError(); // avoid reporting incorrect thread context to Watson
```

Comment:
```suggestion
                // RaiseFailFastException validates that the context matches a valid return address on the stack as part of CET.
                // If the return address is not valid, it rejects the context, flags it as a potential attack and asserts in 
                // checked builds of Windows OS. 
                // Avoid reporting thread context captured by EEPolicy::HandleFatalError since it has IP that does not 
                // match a valid return address on the stack.
                bool fAvoidReportContextToRaiseFailFast = tore.IsFatalError();
```